### PR TITLE
Don't run sandboxing rollback test on mysql

### PR DESCRIPTION
**Pull request by noahmoss**
The test I added in https://github.com/metabase/metabase/pull/44481 seems to now be failing on MySQL 8.0 after merging, even though it passed CI on the PR.

It's a rollback test and I think there's been issues with migration rollback tests failing on MySQL in the past. I don't know the root cause but this PR changes it to only run on postgres/h2. I think this is an OK stopgap to get master green since the test isn't mysql-specific.

### CODE REVIEW

**Comment by noahmoss** [comment]
* **#44515** <a href="https://app.graphite.dev/github/pr/metabase/metabase/44515?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @noahmoss and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: test/metabase/db/schema_migrations_test.clj [hunk 1]
```diff
@@ -2017,33 +2017,35 @@
...
-                                                        :attribute_remappings "{\"foo\", 1}"
-                                                        :permission_id        permission-id}]})
-            expected        {:group_id             1
-                             :table_id             table-id
-                             :attribute_remappings "{\"foo\", 1}"}]
-        (migrate!)
-        (is (=? expected (t2/select-one :sandboxes :table_id table-id)))
-        (migrate! :down 49)
-        (is (=? expected (t2/select-one :sandboxes :table_id table-id)))))))
+  ;; TODO (noahmoss): uncomment when fixed on mysql)
```
Comment (noahmoss) [inline test/metabase/db/schema_migrations_test.clj:2020]:
```suggestion
  ;; TODO (noahmoss): uncomment when fixed on mysql
```

**Review by camsaul — APPROVED** [review]

File: test/metabase/db/schema_migrations_test.clj [hunk 2]
```diff
@@ -2017,33 +2017,35 @@
...
+                                           :values      [{:group_id             1
+                                                          :table_id             table-id
+                                                          :attribute_remappings "{\"foo\", 1}"
+                                                          :permission_id        permission-id}]})
+              expected        {:group_id             1
+                               :table_id             table-id
+                               :attribute_remappings "{\"foo\", 1}"}]
+          (migrate!)
+          (is (=? expected (t2/select-one :sandboxes :table_id table-id)))
+          (migrate! :down 49)
```
Comment (calherries) [inline test/metabase/db/schema_migrations_test.clj:2047]: Not critical, but it should really just exclude MySQL on the rollback e.g.
`(when (not= driver/*driver* :mysql) (migrate! :down 49) ...etc)`

Comment (noahmoss) [reply test/metabase/db/schema_migrations_test.clj:2047]: In this case, the entire point of the test is to test the rollback, so I don't think it matters much

Comment (calherries) [reply test/metabase/db/schema_migrations_test.clj:2047]: True! Ignore my lazy comment

**Pull request merged**